### PR TITLE
Unescape hex/unicode for sample serial data

### PR DESCRIPTION
**Pull request by bberenz**
For #877

### CODE REVIEW

File: sample.html [hunk 1]
```diff
@@ -2021,8 +2021,9 @@ <h4 class="panel-title">Options</h4>
             type: $("input[name='serialType']:checked").val(),
             data: $("#serialCmd").val()
         };
-        //allow some escape characters
-        serialData.data = serialData.data.replace(/\\n/g, "\n").replace(/\\r/g, "\r").replace(/\\t/g, "\t");
+        //allow some escape characters (newlines, tabs, hex/unicode)
+        serialData.data = serialData.data.replace(/\\n/g, "\n").replace(/\\r/g, "\r").replace(/\\t/g, "\t")
+            .replace(/\\[xu]([\dA-Fa-f]+)/g, (m, s1) => String.fromCharCode(parseInt(s1, 16)));
```
Comment (klabarge) [inline sample.html:2026]: This arrow function breaks IE11 support

**Review by klabarge — CHANGES_REQUESTED** [review]

File: sample.html (hunk 1 above)
Comment (klabarge) [reply sample.html:2026]: Can this be rewritten like so?
```js
        serialData.data = serialData.data.replace(/\\n/g, "\n").replace(/\\r/g, "\r").replace(/\\t/g, "\t")
            .replace(/\\[xu]([\dA-Fa-f]+)/g, (m, s1).then(function() {
                String.fromCharCode(parseInt(s1, 16))
            })
        );
```

Comment (bberenz) [reply sample.html]: yes, but i found a case in retesting where the \x codes can try to collect too many values to convert, so I pushed up a change where the function is a variable to be used for both

**Comment by klabarge** [comment]
What data should I be testing?

I'm kind of arbitrarily sending stuff so not sure if it's valid.
#### Test 1
**send:** `hi, serial port\n`
**logs:**`"data":"hi, serial port\n"`

#### Test 2
**send:** `hi, serial port\u`
**logs:**`"data":"hi, serial port\\u"`
**error:** `java.lang.IllegalArgumentException: Less than 4 hex digits in unicode value: '\u' due to end of CharSequence`

**Comment by tresf** [comment]
> What data should I be testing?
> 
> I'm kind of arbitrarily sending stuff so not sure if it's valid.
> 
> #### Test 1
> **send:** `hi, serial port\n` **logs:**`"data":"hi, serial port\n"`
> 
> #### Test 2
> **send:** `hi, serial port\u` **logs:**`"data":"hi, serial port\\u"` **error:** `java.lang.IllegalArgumentException: Less than 4 hex digits in unicode value: '\u' due to end of CharSequence`

Those are pretty good tests.  Some known hex would be nice (newline, tab, a-z stuff, but in hex format).

**Comment by klabarge** [comment]
I sent  hex with new lines and tabs and it seems ok.

```
68 69 0A 09 73 65 72 69 61 6C 0A 70 6F 72 74
```

**Comment by tresf** [comment]
```
thanks
	kyle
merging
```

**Pull request merged**